zerver/lib/queue.py

```diff
@@ -297,12 +297,15 @@ def get_queue_client():
 # randomly close.
 queue_lock = threading.RLock()
 
-def queue_json_publish(queue_name, event, processor):
-    # type: (str, Union[Mapping[str, Any], str], Callable[[Any], None]) -> None
+def queue_json_publish(queue_name, event, processor, call_consume_in_tests=False):
+    # type: (str, Union[Mapping[str, Any], str], Callable[[Any], None], bool) -> None
     # most events are dicts, but zerver.middleware.write_log_line uses a str
     with queue_lock:
         if settings.USING_RABBITMQ:
             get_queue_client().json_publish(queue_name, event)
+        elif call_consume_in_tests:
+            from zerver.worker.queue_processors import get_worker
```

Comment:
Importing this at the top section leads to not-defined-ish errors.

Reply:
Yep, makes sense.  Maybe add a comment explaining that?  Better to put that sort of info in a place it'll be there for future developers :)